pkg/specgen/generate/container.go

```diff
@@ -33,7 +34,40 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 
 		_, mediaType, err := newImage.Manifest(ctx)
 		if err != nil {
-			return nil, err
+			if errors.Cause(err) != image.ErrImageIsBareList {
+				return nil, err
+			}
+			// if err is not runnable image
+			// use the local store image with repo@digest matches with the list, if exists
+			manifestByte, manifestType, err := newImage.GetManifest(ctx, nil)
+			if err != nil {
+				return nil, err
+			}
+			list, err := manifest.ListFromBlob(manifestByte, manifestType)
+			if err != nil {
+				return nil, err
+			}
+			images, err := r.ImageRuntime().GetImages()
+			if err != nil {
+				return nil, err
+			}
+			findLocal := false
+			for _, img := range images {
+				for _, imageDigest := range img.Digests() {
+					for _, listDigest := range list.Instances() {
+						if imageDigest == listDigest {
+							newImage = img
+							s.Image = img.ID()
+							findLocal = true
+							logrus.Debug("image contains manifest list, using image from local storage")
+							break
+						}
+					}
+				}
+			}
```

Comment:
This will match any local image which is also in the list, so it solves the problem when only one of them is present.  If I've pulled more than one using the repository name and their respective digests, this will choose the first one in the list, when the one for the current architecture (if one is present) is listed later in `images`.  Instead of checking for any of the values returned by `list.Instances()`, can we use `list.ChooseInstance()` to select the one that would be correct for the current architecture, and only accept an image that matches the digest it returns?

Reply:
fixed.